task/sql-tasks.js

```diff
@@ -280,7 +465,25 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice as "PricePerItem"
+        FROM Customers
+        JOIN Orders ON Customers.CustomerID=Orders.CustomerID
+        JOIN OrderDetails ON Orders.OrderID=OrderDetails.OrderID
+        JOIN Products ON OrderDetails.ProductID=Products.ProductID
+        WHERE OrderDetails.UnitPrice = (SELECT
+                    MAX(OD.UnitPrice)
+                FROM Customers as C
+                JOIN Orders as O ON O.CustomerID=C.CustomerID
+                JOIN OrderDetails as OD ON OD.OrderID=O.OrderID
+                WHERE Customers.CompanyName=C.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.